src/App.tsx

```diff
@@ -1,17 +1,121 @@
-import React from 'react';
+import React, { useState } from 'react';
+
+import 'bootstrap/dist/css/bootstrap.min.css';
 import './App.css';
 
 import users from './api/users';
+import todos from './api/todos';
+import TodoList from './components/TodoList/TodoList';
+
+const preparedTodos: Todo[] = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId) || null,
+}));
 
 const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [username, setUsername] = useState('');
+  const [titeleValid, setTitleValid] = useState(true);
+  const [usernameValid, setUsernameValid] = useState(true);
+  const [todoList, setTodoList] = useState([...preparedTodos]);
+
+  const addTodo = () => {
+    if (!title) {
+      setTitleValid(false);
+    }
+
+    if (!username) {
+      setUsernameValid(false);
+    }
+
+    if (title && username) {
+      const currentUser = users.find(user => user.name === username) || null;
+      const newTodo = {
+        id: todoList[todoList.length - 1].id + 1,
+        title,
+        userId: currentUser ? currentUser.id : null,
+        completed: false,
+        user: currentUser,
+      };
+
+      setTodoList(prevTodo => [...prevTodo, newTodo]);
+      setTitle('');
+      setUsername('');
+      setTitleValid(true);
+      setUsernameValid(true);
+    }
+  };
+
+  type Event = React.ChangeEvent<HTMLInputElement | HTMLSelectElement>;
```

Comment:
suggestion: types are not usually inserted just between lines, thus either put them at the beginning or move to external file